lib/cocoapods/validator.rb

```diff
@@ -584,8 +584,8 @@ def configure_pod_targets(targets, target_installation_results, deployment_targe
           (build_configuration.build_settings['OTHER_CFLAGS'] ||= '$(inherited)') << ' -Wincomplete-umbrella'
           build_configuration.build_settings['SWIFT_VERSION'] = (pod_target.swift_version || derived_swift_version) if pod_target.uses_swift?
         end
-        pod_target_installation_result.test_specs_by_native_target.each do |test_native_target, test_specs|
-          if pod_target.uses_swift_for_spec?(test_specs.first)
+        pod_target_installation_result.test_specs_by_native_target.each do |test_native_target, test_spec|
+          if pod_target.uses_swift_for_spec?(test_spec)
```

Comment:
is this always a 1 to 1 mapping? as in is there ever a scenario where there would be multiple test specs for a single target?

Reply:
yes it is for non library specs (test specs and app specs)

Reply:
This change right here is the actual crash fix.